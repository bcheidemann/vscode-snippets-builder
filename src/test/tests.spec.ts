import { setupFS } from './helpers/setup-fs';
import { SnippetsBuilder } from '..';
import { join } from 'path';

describe('test', () => {
  it('should generate a snippet file', async () => {
    const fs = setupFS();
    const builder = new SnippetsBuilder({
      fs,
      srcDir: join(__dirname, 'src'),
    });
    await builder.build();
    expect(fs.files).toMatchSnapshot();
  })
});
