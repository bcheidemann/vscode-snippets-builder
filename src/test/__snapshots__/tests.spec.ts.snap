// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should generate a snippet file 1`] = `
Object {
  "/home/ben/Code/vscode-snippets-builder/src/test/helpers/src/class.template": "class MyClass {
  \${0}
}",
  "/home/ben/Code/vscode-snippets-builder/src/test/helpers/src/composite.snippet.json": "{
  \\"0\\": {
    \\"0\\": {
      \\"0\\": \\"'Hello World!'\\",
      \\"template\\": \\"./body.template\\"
    },
    \\"template\\": \\"./method.template\\"
  },
  \\"name\\": \\"Class with Method\\",
  \\"alias\\": \\".myclassmethod\\",
  \\"template\\": \\"./class.template\\"
}",
  "/home/ben/Code/vscode-snippets-builder/src/test/helpers/src/log.template": "console.log(\${0})",
  "/home/ben/Code/vscode-snippets-builder/src/test/helpers/src/method.template": "  public myMethod() { \${0} }",
}
`;
